### ISSUE TITLE
Javadoc on Module property should be able to define special default value

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#14178]
detected at: 

CustomImportOrderCheck.customImportOrderRules has very special class field that defined it default value.
https://github.com/checkstyle/checkstyle/blob/3ba30a040638c3ff14c44676a2f019930bfaf827/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java#L349-L358

https://github.com/checkstyle/checkstyle/blob/3ba30a040638c3ff14c44676a2f019930bfaf827/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java#L260-L261

usage of mactros results in:
`    -Property {@code customImportOrderRules} - Specify format of order declaration customizing by user. Type is {@code java.lang.String}. Default value is {@code null}.`
`    +Property {@code customImportOrderRules} - Specify format of order declaration customizing by user. Type is {@code java.lang.String}. Default value is {@code ""}.`

we need to remove temporal workaround with hardcoded default value and find way to make it declarative in javadoc of setter.

**Comment by romani** [comment]
my bad ..... `{@code ""}` is what macros demands and it is right , it is a value of DEFAULT_CUSTOM_IMPORT_ORDER_RULES

**Comment by romani** [comment]
One of the approach to address this is to make new javadoc tag:
`@defaultvalue DEFAULT_CUSTOM_IMPORT_ORDER_RULES`

**Comment by relentless-pursuit** [comment]
I am on it.

**Comment by romani** [comment]
@relentless-pursuit, please beware that issue is not approved, so please send draft PR to let other maintenance agree on design and you will not waste time on redoing

**Comment by relentless-pursuit** [comment]
Sure

**Comment by relentless-pursuit** [comment]
@Romani, any reference PR for creating custom Javadocs tags?

**Comment by romani** [comment]
Look at history  commit of this line https://github.com/checkstyle/checkstyle/blob/d7f97f34021bab6e5027a947eabbcd40300d4567/pom.xml#L531

**Comment by romani** [comment]
@relenless-pursuit, are you doing something on this issue ?

@rnveach, @nrmancuso, lets agree on name of javadoc tag, see https://github.com/checkstyle/checkstyle/issues/14178#issuecomment-1868553955 . It might be useful more widely as we start removal of javadoc for all propertties from content of javadoc over class.

**Comment by rnveach** [comment]
I don't agree with this issue.
https://github.com/checkstyle/checkstyle/pull/14180#pullrequestreview-1795508066

**Comment by relentless-pursuit** [comment]
> @relentless-pursuit, are you doing something on this issue ?
> 
> @rnveach, @nrmancuso, lets agree on name of javadoc tag, see [#14178 (comment)](https://github.com/checkstyle/checkstyle/issues/14178#issuecomment-1868553955) . It might be useful more widely as we start removal of javadoc for all propertties from content of javadoc over class.

I actually started doing it but maybe in the wrong direction. I was trying to create a DefaultValue named Check file. However, after giving me the reference links, i am prcoeeding in the direction of `@noinspectionreason`, they way it was created. I am giving myself a deadline of 4 days to do a draft PR. But, if it is a priority and needs immediate attention, i will pass it on.

**Comment by relentless-pursuit** [comment]
And, after understanding the issue, i don't know if there is any other solution. For instance, i encountered this issue while completing the PR for SuppressedWarnings. And, @romani, futher improved it in subequent migration handling the issue of `null`. If there is any other solution or direction where we could invesitgate, pleae let me know.

**Comment by nrmancuso** [comment]
I also do not agree on the premise of this issue.

>Javadoc on Module property should be able to define special default value

If we are not able to readily generate documentation from a given check, this is a clear indication that it is an issue with the check itself. Creating some annotation. etc, seems like more work than just "fixing" the check.

Maybe I am missing some detail, but it seems pretty straightforward to:
1. Create class field of type string
2. Set this field via the setter, with no transformation
3. Override AbstractCheck#init to handle transformation and populate `importToGroupList`
4. Use this list in the check as before

**Comment by rnveach** [comment]
The other options I mentioned in the PR I referenced is to override the code that says to use `null` in this scenario and switch it to blanks (current xdoc default) since it is said `null` is never a valid configuration value, or rename the field to match the setter so it can identify the default and handle it that way.

**Comment by romani** [comment]
>Set this field via the setter, with no transformation

It will be problem for pitest.

>since it is said null is never a valid configuration value

In many cases it is valid for java bean mode.

>Maybe I am missing some detail, but it seems pretty straightforward to

As we don't understand each other, and it was only me doing migration. Can you do this migration?

**Comment by nrmancuso** [comment]
>If we are not able to readily generate documentation from a given check, this is a clear indication that it is an issue with the check itself.

I have done some investigation and opened an issue regarding check property design: https://github.com/checkstyle/checkstyle/issues/14216

I am good to move ahead with documentation generation PR, with `// until https://github.com/checkstyle/checkstyle/issues/14216` comments on hacks.

**Comment by romani** [comment]
@rnveach , looks like more maintainers see possibility to move Check update to separate issue and unblock macros PR, we disagree on implementation of Check update.
I can open new issue to simply request resolvement of hack, no specific details on how to do it, reference it in macros PR, merge it.

Fix in module, will be results of conversations in this or https://github.com/checkstyle/checkstyle/issues/14216 or new issue.

Does it work for you?

**Comment by rnveach** [comment]
I am for less hacks, so I see it better we just do the fix and convert all in one, similar to other macro PRs with text updates, then implementing this issue or the other PR.

**Comment by romani** [comment]
we can not find a way to agree on this, so I am closing this issue, we will do update in Check to make better type for problematic property - https://github.com/checkstyle/checkstyle/issues/14216
